public/dorequest.php

```diff
@@ -277,6 +345,104 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
         $response['AchievementID'] = $achIDToAward;
         break;
 
+    // This is only currently supported for "Standalone" integrations.
+    case "awardachievements":
+        if (request()->method() !== 'POST') {
+            return DoRequestError('Access denied.', 405, 'access_denied');
+        }
+
+        $achievementIdsInput = request()->input('a', '');
+        $hardcore = (bool) request()->input('h', 0);
+        $validationHash = request()->input('v');
```

Comment:
Recommend changing these `request()->input` calls to `request->post()` to ensure the data is in the POST body and not the query parameters.

Reply:
Not a problem, changed and test cases revised.